crates/biome_js_analyze/src/lint/nursery/use_at_index.rs

```diff
@@ -0,0 +1,927 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource, RuleSourceKind,
+};
+use biome_console::{markup, MarkupBuf};
+use biome_js_factory::make::{self};
+use biome_js_syntax::{
+    AnyJsCallArgument, AnyJsExpression, AnyJsLiteralExpression, JsBinaryExpression,
+    JsCallExpression, JsComputedMemberExpression, JsParenthesizedExpression,
+    JsStaticMemberExpression, JsUnaryExpression, T,
+};
+use biome_rowan::{declare_node_union, AstNode, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+
+declare_lint_rule! {
+    /// Enforce using .at to retrieve elements.
+    ///
+    /// When extracting elements from an array, especially when retrieving from the end, `.at` is convenient. Replace the previously used syntax with `.at()`.
+    ///
+    /// ## Options
+    ///
+    /// ### `checkAllIndexAccess`
+    ///
+    /// By default, only negative element accesses will use errors, but I will also generate errors for positive accesses.
+    ///
+    /// ```json,ignore
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "checkAllIndexAccess": true
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = array[array.length - 1];
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = array[array.length - 5];
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = array.slice(-1)[0];
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = array.slice(-1).pop();
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = array.slice(-5).shift();
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = string.charAt(string.length - 5);
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = lodash.last(array);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const foo = array.at(-1);
+    /// ```
+    ///
+    /// ```js
+    /// const foo = array.at(-5);
+    /// ```
+    ///
+    /// ```js
+    /// const foo = array[100];
+    /// ```
+    ///
+    /// ```js
+    /// const foo = array.at(array.length - 1);
+    /// ```
+    ///
+    /// ```js
+    /// array[array.length - 1] = foo;
+    /// ```
+    pub UseAtIndex {
+        version: "next",
+        name: "useAtIndex",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintUnicorn("prefer-at")],
+        source_kind: RuleSourceKind::Inspired,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+/// If the node is a parenthized expression, it returns the expression inside.
+/// # Examples
+/// ```js
+///    a // Some(a)
+///    (a) // Some(a)
+///    (a + b) // Some(a + b)
+/// ```
+fn solve_parenthesized_expression(mut node: AnyJsExpression) -> Option<AnyJsExpression> {
```

Comment:
This is just my personal suggestion: 
consider using a name like `unwrap_parenthesized_expression` or `strip_parentheses`. These names might more clearly describe the action of removing or unwrapping parentheses from an expression, which could be more intuitive in the context of AST manipulation?

What do you think?